llvm/lib/Target/AArch64/GISel/AArch64CallLowering.cpp

```diff
@@ -207,14 +207,14 @@ struct IncomingArgHandler : public CallLowering::IncomingValueHandler {
   /// How the physical register gets marked varies between formal
   /// parameters (it's a basic-block live-in), and a call instruction
   /// (it's an implicit-def of the BL).
-  virtual void markPhysRegUsed(MCRegister PhysReg) = 0;
+  virtual void markPhysRegUsed(Register PhysReg) = 0;
 };
 
 struct FormalArgHandler : public IncomingArgHandler {
   FormalArgHandler(MachineIRBuilder &MIRBuilder, MachineRegisterInfo &MRI)
       : IncomingArgHandler(MIRBuilder, MRI) {}
 
-  void markPhysRegUsed(MCRegister PhysReg) override {
+  void markPhysRegUsed(Register PhysReg) override {
     MIRBuilder.getMRI()->addLiveIn(PhysReg);
```

Comment:
I think the confusion here is deeper, and I'm not sure a virtual register should reach here. addLiveIn for a virtual register also doesn't make sense.

Reply:
It's not a virtual register when FormalArgHandler::markPhysRegUsed is called. It's only virtual when ReturnedArgCallReturnHandler::markPhysRegUsed is called. This is a virtual function.

Reply:
I've changed to use Register::asMCReg() instead of implicit conversion for the addLiveIn calls. That will check that it is safe to convert from Register to MCRegister.